src/zarr/core/metadata/v2.py

```diff
@@ -332,6 +367,18 @@ def parse_fill_value(fill_value: object, dtype: np.dtype[Any]) -> Any:
             raise ValueError(
                 f"fill_value {fill_value!r} is not valid for dtype {dtype}; must be a unicode string"
             )
+    elif dtype.kind in "SV" and isinstance(fill_value, str):
```

Comment:
All of this logic is the same as new lines added above. Would it be possible to deduplicate it somehow?

Reply:
I assume you're referring to the handling in `parse_structured_fill_value`. There are slight differences between these cases that make deduplication seem pretty difficult to my mind. In the structured case, what's being handled is a potentially serialized (to string) structured fill value (hence the next step which takes that and constructs the compound value): `fill_value = np.frombuffer(decoded, dtype=dtype)[0]`.